pkgs/applications/editors/emacs/generic.nix

```diff
@@ -75,9 +75,9 @@ in stdenv.mkDerivation {
           "${lib.getLib stdenv.cc.libc}/lib"
 
           # Executable paths necessary for compilation (ld, as):
-          "${lib.getBin stdenv.cc.cc}"
-          "${lib.getBin stdenv.cc.bintools}"
-          "${lib.getBin stdenv.cc.bintools.bintools}"
+          "${lib.getBin stdenv.cc.cc}/bin"
+          "${lib.getBin stdenv.cc.bintools}/bin"
+          "${lib.getBin stdenv.cc.bintools.bintools}/bin"
```

Comment:
In my tests, it looked like libgccjit would search through subdirs for binaries. At least that's what I was led to believe based on the `prefix` name in [the docs](https://gcc.gnu.org/onlinedocs/gcc/Directory-Options.html). Weirdly enough, it can find the `as` binaries on my macOS system, and I was pretty certain we tested gccemacs's ability to compile on a Linux machine also, with -B paths sans /bin.

I'm pretty sure your change results in the right directory names, but I'm a bit worried we may be flip-flopping between two solutions that only half fix the problem. Hrm. I can try this change on macOS later this weekend, can check if it continues working.

Reply:
Hm... I read those docs differently than you, but I can certainly see the confusion. Give the `/bin` version a shot to see if it works for you, too. In the overlay issue I linked, you can see that another user was pulling `as` in via another mechanism, and it's not hard to imagine some `as` getting onto `$PATH` through various out-of-band mechanisms.